### PR TITLE
feat(eslint-plugin): added new plugin

### PR DESCRIPTION
**Pull request by denisbalyko**
Helps not to write complex constructs and function calls inside className.

https://nda.ya.ru/3Vo6KU

### CODE REVIEW

File: packages/eslint-plugin/lib/rules/no-classname-runtime.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  meta: {
+    docs: {
+      description: 'Do not use the @bem-react/classname function in runtime code',
+      category: 'perf',
+      recommended: true,
+    },
+  },
+
+  create: function(context) {
+    function isCanBeSimplified(arg) {
```
Comment (victor-homyakov) [inline packages/eslint-plugin/lib/rules/no-classname-runtime.js]: Strange naming, "canBeSimplified" looks better

File: packages/eslint-plugin/lib/rules/no-classname-runtime.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+      if (arg && (arg.type === 'Literal' || arg.type === 'Identifier')) return true
+
+      if (arg.type === 'MemberExpression') {
```
Comment (victor-homyakov) [inline packages/eslint-plugin/lib/rules/no-classname-runtime.js]: `switch (arg.type)` works better for that amount of `if`s

File: packages/eslint-plugin/lib/rules/no-classname-runtime.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+      }
+
+      if (arg.type === 'BinaryExpression' || arg.type === 'LogicalExpression') {
+        return isCanBeSlow([arg.left, arg.right])
+      }
+
+      return false
+    }
+
+    function isCanBeSlow(args, dbg) {
```
Comment (victor-homyakov) [inline packages/eslint-plugin/lib/rules/no-classname-runtime.js]: `canBeSlow` or `isPotentiallySlow`

Comment (victor-homyakov) [inline packages/eslint-plugin/lib/rules/no-classname-runtime.js]: Unused parameter `dbg`

File: packages/eslint-plugin/lib/rules/no-classname-runtime.js [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    }
+
+    return {
+      JSXAttribute: function(node) {
+        const isEvaluatedClassName =
+          node.name.type === 'JSXIdentifier' &&
+          node.name.name === 'className' &&
+          node.value.type === 'JSXExpressionContainer'
+
+        if (isEvaluatedClassName) {
```
Comment (victor-homyakov) [inline packages/eslint-plugin/lib/rules/no-classname-runtime.js]: You may rewrite it in this way to reduce the amount of nested code

```js
if (!isCanBeSimplified) return;

// ...
```

File: packages/eslint-plugin/lib/rules/no-classname-runtime.js [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+  create: function(context) {
+    function canBeSimplified(arg) {
+      if (!arg) return false
+
+      switch (arg.type) {
+        case 'Literal':
+        case 'Identifier':
+          return true
+        case 'MemberExpression':
+          // Exception for this.props.x or this.state.x
```
[the remark this answers is not in the document]
Comment (denisbalyko) [reply packages/eslint-plugin/lib/rules/no-classname-runtime.js]: They need to be taken out on top.

**Pull request merged**